test/Api.IntegrationTest/SecretsManager/Controllers/ServiceAccountsControllerTests.cs

```diff
@@ -73,51 +77,94 @@ public async Task ListByOrganization_SmAccessDenied_NotFound(bool useSecrets, bo
 
-        // Setup access for two
-        var accessPolicies = serviceAccountIds.Take(2).Select(
-            id => new UserServiceAccountAccessPolicy
+        Assert.NotNull(result);
+        Assert.NotEmpty(result.Data);
+        Assert.Equal(serviceAccountIds.Count, result.Data.Count());
+
+        foreach (var item in expectedAccess)
+        {
+            var serviceAccountResult = result.Data.FirstOrDefault(x => x.Id == item.Key);
+            if (serviceAccountResult != null)
             {
-                OrganizationUserId = orgUser.Id,
-                GrantedServiceAccountId = id,
-                Read = true,
-                Write = false,
-            }).Cast<BaseAccessPolicy>().ToList();
+                Assert.Equal(item.Value, serviceAccountResult.AccessToSecrets);
+            }
+        }
+    }
 
-        await _accessPolicyRepository.CreateManyAsync(accessPolicies);
+    [Theory]
+    [InlineData(false)]
+    [InlineData(true)]
+    public async Task ListByOrganization_UserPartialAccess_ReturnsServiceAccountsUserHasAccessTo(
+        bool includeAccessToSecrets)
+    {
+        var (orgId, serviceAccountIds) =
+            await SetupListByOrganizationRequestAsync(PermissionType.RunAsUserWithPermission);
+        var expectedAccess = await SetupServiceAccountSecretAccessAsync(serviceAccountIds, orgId);
 
-        var response = await _client.GetAsync($"/organizations/{org.Id}/service-accounts");
+        var serviceAccountWithoutAccess = await _serviceAccountRepository.CreateAsync(new ServiceAccount
+        {
+            OrganizationId = orgId,
+            Name = _mockEncryptedString
+        });
+
+        var response =
+            await _client.GetAsync(
+                $"/organizations/{orgId}/service-accounts?includeAccessToSecrets={includeAccessToSecrets}");
         response.EnsureSuccessStatusCode();
-        var result = await response.Content.ReadFromJsonAsync<ListResponseModel<ServiceAccountResponseModel>>();
+        var result = await response.Content
+            .ReadFromJsonAsync<ListResponseModel<ServiceAccountSecretsDetailsResponseModel>>();
 
         Assert.NotNull(result);
         Assert.NotEmpty(result.Data);
-        Assert.Equal(2, result.Data.Count());
+        Assert.Equal(serviceAccountIds.Count, result.Data.Count());
+        Assert.DoesNotContain(result.Data, x => x.Id == serviceAccountWithoutAccess.Id);
+
+        if (includeAccessToSecrets)
+        {
+            foreach (var item in expectedAccess)
+            {
+                var serviceAccountResult = result.Data.FirstOrDefault(x => x.Id == item.Key);
+                if (serviceAccountResult != null)
+                {
+                    Assert.Equal(item.Value, serviceAccountResult.AccessToSecrets);
+                }
+            }
+        }
+        else
+        {
+            Assert.DoesNotContain(result.Data, x => x.AccessToSecrets != 0);
```

Comment:
⛏️ i thin; it's better to not use double negation, easier to read like below:
```suggestion
            Assert.Contains(result.Data, x => x.AccessToSecrets == 0);
```

Reply:
Updated in https://github.com/bitwarden/server/pull/4156/commits/c9e5d972170e1507c708060cf085e3a252d497a6